docs/radar.js

```diff
@@ -347,18 +353,65 @@ function radar_visualization(config) {
                  return (d.link && config.links_in_new_tabs) ? "_blank" : null;
               })
             .append("text")
-              .attr("transform", function(d, i) { return legend_transform(quadrant, ring, config.legend_column_width, i); })
+              .attr("transform", function(d, i) { return legendTransform(quadrant, ring, config.legend_column_width, i, previousLegendHeight); })
               .attr("class", "legend" + quadrant + ring)
               .attr("id", function(d, i) { return "legendItem" + d.id; })
-              .text(function(d, i) { return d.id + ". " + d.label; })
+              .text(function(d) { return d.id + ". " + d.label; })
               .style("font-family", config.font_family)
               .style("font-size", "11px")
               .on("mouseover", function(d) { showBubble(d); highlightLegendItem(d); })
-              .on("mouseout", function(d) { hideBubble(d); unhighlightLegendItem(d); });
+              .on("mouseout", function(d) { hideBubble(d); unhighlightLegendItem(d); })
+              .call(wrapText)
+              .each(function() {
+                previousLegendHeight += d3.select(this).node().getBBox().height;
+              });
       }
     }
   }
 
+  function wrapText(text) {
+    let heightForNextElement = 0;
+
+    text.each(function() {
+      const textElement = d3.select(this);
+      const words = textElement.text().split(" ");
+      let line = [];
+
+      // Use '|' at the end of the string so that spaces are not trimmed during rendering.
+      const number = `${textElement.text().split(".")[0]}. |`;
+      const legendNumberText = textElement.append("tspan").text(number);
+      const legendBar = textElement.append("tspan").text('|');
+      const numberWidth = legendNumberText.node().getComputedTextLength() - legendBar.node().getComputedTextLength();
+
+      textElement.text(null);
+
+      let tspan = textElement
+          .append("tspan")
+          .attr("x", 0)
+          .attr("y", heightForNextElement)
+          .attr("dy", 0);
+
+      for (let i = 0; i < words.length; i++) {
+        line.push(words[i]);
+        tspan.text(line.join(" "));
+
+        if (tspan.node().getComputedTextLength() > config.legend_column_width) {
```

Comment:
```suggestion
        if (tspan.node().getComputedTextLength() > config.legend_column_width && i != 1) {
```

Reply:
I believe, the newline logic should not be invoked if we're processing the "first" word with index 1 (given `word[0]` is the number). May be worth adding a comment. When faced with a single very long word, we end up with a situation where there is a line added with just the number, which looks odd.

<img width="553" alt="image" src="https://github.com/user-attachments/assets/29659979-1545-4a5c-98ad-36b649345980" />

```svg
<a href="#">
 <text transform="translate(450,-150.59999656677246)" class="legend31" id="legendItem34" style="font-family: &quot;Courier New&quot;, Arial, Helvetica; font-size: 11px;">
  <tspan x="0" y="0" dy="0">34.</tspan>
  <tspan x="26.403123378753662" dy="10">AWSAthenaAthenaAthenaAthena</tspan>
 </text>
</a>
```

Reply:
Very good point. I added the condition and a comment to highlight the edge case. I don't think we have many other option here apart from skipping the text of the first line for now. Thanks for point out this case 🙏